src/world_new/views/WorldDataView.cpp

```diff
@@ -0,0 +1,158 @@
+//
+// Created by john on 1/6/20.
+//
+
+#include "world_new/views/WorldDataView.hpp"
+
+#include "world_new/Ball.hpp"
+#include "world_new/WorldData.hpp"
+
+namespace rtt::world_new::view {
+
+std::vector<view::RobotView> const &WorldDataView::getUs() const noexcept { return data->getUs(); }
+
+std::vector<view::RobotView> const &WorldDataView::getThem() const noexcept { return data->getThem(); }
+
+std::vector<robot::Robot> const &WorldDataView::getRobots() const noexcept { return data->getRobots(); }
+
+std::optional<view::BallView> WorldDataView::getBall() const noexcept { return data->getBall(); }
+
+std::optional<view::RobotView> WorldDataView::getRobotForId(uint8_t id, bool ourTeam) const noexcept {
+    auto const &_data = ourTeam ? getUs() : getThem();  // const&, prevents copy
+
+    // return if the lambda evaluates to true
+    auto _rbt = std::find_if(_data.begin(), _data.end(), [id](RobotView rbt) { return rbt->getId() == id; });
+
+    // not found ? None, else Some(*_robot);
+    return _rbt == _data.end() ? std::optional<RobotView>() : *_rbt;
+}
```

Comment:
This looks good. This is one of the functions you maaayybee want to benchmark as it's called a lot internally.

Reply:
I'll benchmark in a bit :)